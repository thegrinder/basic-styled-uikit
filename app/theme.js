const theme = {
  uiKit: {
    buttons: {
      btnTypes: {
        default: {
          regular: {
            color: '#222',
            bgColor: 'transparent',
            borderColor: '#e5e5e5',
          },
          hover: {
            color: '#222',
            bgColor: 'transparent',
            borderColor: '#b2b2b2',
          },
          pressed: {
            color: '#222',
            bgColor: 'transparent',
            borderColor: '#999',
          },
        },
        primary: {
          regular: {
            color: '#fff',
            bgColor: '#1e87f0',
            borderColor: 'transparent',
          },
          hover: {
            color: '#fff',
            bgColor: '#0f7ae5',
            borderColor: 'transparent',
          },
          pressed: {
            color: '#fff',
            bgColor: '#0e6dcd',
            borderColor: 'transparent',
          },
        },
        secondary: {
          regular: {
            color: '#fff',
            bgColor: '#222',
            borderColor: 'transparent',
          },
          hover: {
            color: '#fff',
            bgColor: '#151515',
            borderColor: 'transparent',
          },
          pressed: {
            color: '#fff',
            bgColor: '#080808',
            borderColor: 'transparent',
          },
        },
        danger: {
          regular: {
            color: '#fff',
            bgColor: '#f0506e',
            borderColor: 'transparent',
          },
          hover: {
            color: '#fff',
            bgColor: '#ee395b',
            borderColor: 'transparent',
          },
          pressed: {
            color: '#fff',
            bgColor: '#ec2147',
            borderColor: 'transparent',
          },
        },
      },
      btnSizes: {
        small: {
          padding: '0 15px',
          lineHeight: '28px',
          fontSize: '14px',
        },
        default: {
          padding: '0 30px',
          fontSize: '14px',
          lineHeight: '38px',
        },
        large: {
          padding: '0 40px',
          lineHeight: '53px',
          fontSize: '16px',
        },
      },
    },
  },
};

export default theme;
