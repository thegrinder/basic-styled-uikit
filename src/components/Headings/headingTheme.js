import { rem } from '../../helpers/utils';

const headingTheme = {
  colours: {
    white: '#fff',
    lightest: '#ccc',
    light: '#999',
    normal: '#666',
    dark: '#424242',
    primary: '#1e87f0',
    success: '#32d296',
    warning: '#faa05a',
    danger: '#f0506e',
  },
  sizings: {
    h1: rem(96),
    h2: rem(80),
    h3: rem(48),
    h4: rem(36),
    h5: rem(24),
    h6: rem(20),
  },
};

export default headingTheme;
