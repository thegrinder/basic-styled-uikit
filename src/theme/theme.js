import button from '../components/Button/buttonTheme';
import link from '../components/Link/linkTheme';
import form from '../components/Form/formTheme';
import text from '../components/Text/textTheme';
import heading from '../components/Headings/headingTheme';
import colorBox from '../components/ColorBox/colorBoxTheme';

const theme = {
  uiKit: {
    button,
    link,
    form,
    text,
    heading,
    colorBox,
  },
};

export default theme;
