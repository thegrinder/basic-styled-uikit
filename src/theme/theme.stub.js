import { rem } from 'basic-styled-uikit';

const colors = {
  transparent: 'transparent',
  white: '#fff',
  gray50: '#fafafa',
  gray100: '#f8f9fa',
  gray200: '#e9ecef',
  gray300: '#dee2e6',
  gray400: '#ced4da',
  gray500: '#adb5bd',
  gray600: '#868e96',
  gray700: '#495057',
  gray800: '#343a40',
  gray900: '#212529',
  black: '#161616',

  primary50: '#e3f2fd',
  primary100: '#bbdefb',
  primary200: '#90caf9',
  primary300: '#64b5f6',
  primary400: '#42a5f5',
  primary500: '#2196f3',
  primary600: '#1e88e5',
  primary700: '#1976d2',
  primary800: '#1565c0',
  primary900: '#0d47a1',

  success50: '#e8f5e9',
  success100: '#c8e6c9',
  success200: '#a5d6a7',
  success300: '#81c784',
  success400: '#66bb6a',
  success500: '#4caf50',
  success600: '#43a047',
  success700: '#388e3c',
  success800: '#2e7d32',
  success900: '#1b5e20',

  warning50: '#fffde7',
  warning100: '#fff9c4',
  warning200: '#fff59d',
  warning300: '#fff176',
  warning400: '#ffee58',
  warning500: '#ffeb3b',
  warning600: '#fdd835',
  warning700: '#fbc02d',
  warning800: '#f9a825',
  warning900: '#f57f17',

  danger50: '#ffebee',
  danger100: '#ffcdd2',
  danger200: '#ef9a9a',
  danger300: '#e57373',
  danger400: '#ef5350',
  danger500: '#f44336',
  danger600: '#e53935',
  danger700: '#d32f2f',
  danger800: '#c62828',
  danger900: '#b71c1c',
};

const theme = {
  uiKit: {
    colors,
    // button theme
    button: {
      btnTypes: {
        basic: {
          normal: {
            color: colors.gray900,
            backgroundColor: colors.transparent,
            borderColor: colors.gray300,
          },
          hover: {
            color: colors.gray900,
            backgroundColor: colors.transparent,
            borderColor: colors.gray400,
          },
          active: {
            color: colors.gray900,
            backgroundColor: colors.transparent,
            borderColor: colors.gray500,
          },
        },
        default: {
          normal: {
            color: colors.black,
            backgroundColor: colors.gray300,
            borderColor: colors.transparent,
          },
          hover: {
            color: colors.black,
            backgroundColor: colors.gray400,
            borderColor: colors.transparent,
          },
          active: {
            color: colors.black,
            backgroundColor: colors.gray500,
            borderColor: colors.transparent,
          },
        },
        primary: {
          normal: {
            color: colors.white,
            backgroundColor: colors.primary500,
            borderColor: colors.transparent,
          },
          hover: {
            color: colors.white,
            backgroundColor: colors.primary600,
            borderColor: colors.transparent,
          },
          active: {
            color: colors.white,
            backgroundColor: colors.primary700,
            borderColor: colors.transparent,
          },
        },
        danger: {
          normal: {
            color: colors.white,
            backgroundColor: colors.danger500,
            borderColor: colors.transparent,
          },
          hover: {
            color: colors.white,
            backgroundColor: colors.danger600,
            borderColor: colors.transparent,
          },
          active: {
            color: colors.white,
            backgroundColor: colors.danger700,
            borderColor: colors.transparent,
          },
        },
      },
      btnSizes: {
        small: {
          padding: `0 ${rem(15)}`,
          lineHeight: rem(28),
          fontSize: rem(14),
        },
        default: {
          padding: `0 ${rem(30)}`,
          fontSize: rem(16),
          lineHeight: rem(38),
        },
        large: {
          padding: `0 ${rem(40)}`,
          lineHeight: rem(53),
          fontSize: rem(22),
        },
      },
      common: {
        fontWeight: '700',
        letterSpacing: '.4px',
      },
    },
    // form theme
    form: {
      regular: {
        states: {
          valid: {
            normal: {
              backgroundColor: colors.white,
              borderColor: colors.gray300,
              color: colors.gray700,
            },
            active: {
              backgroundColor: colors.white,
              borderColor: colors.primary500,
              color: colors.gray700,
            },
            disabled: {
              backgroundColor: colors.gray100,
              borderColor: colors.gray300,
              color: colors.gray600,
            },
          },
          invalid: {
            backgroundColor: colors.white,
            color: colors.danger500,
            borderColor: colors.danger500,
          },
        },
        common: {},
      },
      inline: {
        states: {
          valid: {
            normal: {
              backgroundColor: colors.transparent,
              borderColor: colors.transparent,
              color: colors.gray700,
            },
            hover: {
              backgroundColor: colors.transparent,
              borderColor: colors.gray300,
              color: colors.gray700,
            },
            active: {
              backgroundColor: colors.transparent,
              borderColor: colors.primary500,
              color: colors.gray700,
            },
            disabled: {
              backgroundColor: colors.gray100,
              borderColor: colors.gray300,
              color: colors.gray600,
            },
          },
          invalid: {
            backgroundColor: colors.transparent,
            color: colors.danger500,
            borderColor: colors.danger500,
          },
        },
        common: {},
      },
      // radio and checkbox theme
      misc: {
        states: {
          valid: {
            normal: {
              borderColor: colors.gray300,
              backgroundColor: colors.transparent,
            },
            active: {
              borderColor: colors.primary500,
              backgroundColor: colors.transparent,
            },
            checked: {
              borderColor: colors.transparent,
              backgroundColor: colors.primary500,
            },
          },
          invalid: {
            borderColor: colors.danger500,
            backgroundColor: colors.transparent,
          },
        },
        common: {},
      },
      // toggle theme
      toggle: {
        normal: {
          backgroundColor: colors.gray400,
        },
        checked: {
          backgroundColor: colors.primary500,
        },
      },
    },
    text: {
      sizings: {
        small: rem(14),
        medium: rem(16),
        large: rem(20),
      },
    },
    heading: {
      sizings: {
        h1: rem(36),
        h2: rem(32),
        h3: rem(28),
        h4: rem(24),
        h5: rem(20),
        h6: rem(18),
      },
    },
  },
};

export default theme;
