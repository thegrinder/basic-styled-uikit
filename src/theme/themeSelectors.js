export { getBoxBgColor, getBoxBorderColor } from '../components/ColorBox/colorBoxHelpers';
export { getHeadingSizing, getHeadingColor } from '../components/Headings/headingHelpers';
export { getTextSizing, getTextColor } from '../components/Text/textHelpers';
