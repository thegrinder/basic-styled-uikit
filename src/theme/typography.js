export const getBaseFontSize = ({ theme }) => theme.uiKit.typography.baseFontSize;

export const getBaseLineHeight = ({ theme }) => theme.uiKit.typography.baseLineHeight;

export const calcLineHeight = () => {};

const typography = {
  baseFontSize: 16,
  baseLineHeight: 1.5,
};

export default typography;
