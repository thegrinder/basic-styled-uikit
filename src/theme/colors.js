import { getColors } from './themeSelectors';

export const getColor = color => ({ theme }) => getColors(theme)[color];

const colors = {
  neutral0: '#fff',
  neutral100: '#f8f9fa',
  neutral200: '#e9ecef',
  neutral300: '#dee2e6',
  neutral400: '#ced4da',
  neutral500: '#adb5bd',
  neutral600: '#868e96',
  neutral700: '#495057',
  neutral800: '#343a40',
  neutral900: '#212529',
  neutral1000: '#161616',

  primary200: '#90caf9',
  primary500: '#2196f3',
  success200: '#a5d6a7',
  success500: '#4caf50',
  warning200: '#fff59d',
  warning500: '#ffeb3b',
  danger200: '#ef9a9a',
  danger500: '#f44336',
};

export default colors;
