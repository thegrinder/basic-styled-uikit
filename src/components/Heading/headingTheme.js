import colors from '../../theme/colors/colorsTheme';

const headingTheme = {
  sizings: {
    h1: {
      fontSize: '3rem',
    },
    h2: {
      fontSize: '2.5rem',
    },
    h3: {
      fontSize: '2rem',
    },
    h4: {
      fontSize: '1.5rem',
    },
    h5: {
      fontSize: '1.25rem',
    },
    h6: {
      fontSize: '1rem',
    },
  },
  colors: {
    white: colors.white,
    light: colors.gray700,
    default: colors.gray800,
    dark: colors.gray900,
    black: colors.black,

    primary: colors.primary500,
    success: colors.success500,
    warning: colors.warning500,
    danger: colors.danger500,
  },
  common: {
    fontWeight: 700,
    fontFamily: 'Lato, Arial, Helvetica, sans-serif',
  },
};

export default headingTheme;
