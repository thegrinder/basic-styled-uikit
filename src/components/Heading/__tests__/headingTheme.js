import theme from '../../../theme/theme';
import headingTheme, {
  getHeadingSizingStyle,
  getHeadingColor,
  getHeadingCommonStyle,
} from '../headingTheme';

const {
  sizings,
  colors,
  common,
} = headingTheme;

describe('headingTheme selectors', () => {
  describe('getHeadingSizingStyle', () => {
    Object.keys(sizings).forEach((sizing) => {
      it('should return the correct size style object', () => {
        expect(JSON.stringify(getHeadingSizingStyle(theme, sizing)))
          .toEqual(JSON.stringify(sizings[sizing]));
      });
    });
  });
  describe('getHeadingColor', () => {
    Object.keys(colors).forEach((color) => {
      it('should return the correct color', () => {
        expect(getHeadingColor(theme, color)).toEqual(colors[color]);
      });
    });
  });

  describe('getHeadingCommonStyle', () => {
    it('should return the correct common style object', () => {
      expect(JSON.stringify(getHeadingCommonStyle(theme)))
        .toEqual(JSON.stringify(common));
    });
  });
});
