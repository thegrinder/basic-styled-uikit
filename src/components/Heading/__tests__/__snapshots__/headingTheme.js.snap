// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`headingTheme should be defined 1`] = `
Object {
  "colors": Object {
    "danger": "#f44336",
    "neutral1000": "#161616",
    "neutral600": "#868e96",
    "neutral800": "#343a40",
    "primary": "#2196f3",
    "success": "#4caf50",
    "warning": "#ffeb3b",
  },
  "common": Object {
    "fontFamily": "Lato, Arial, Helvetica, sans-serif",
    "fontWeight": 700,
  },
  "sizings": Object {
    "h1": Object {
      "fontSize": "3rem",
    },
    "h2": Object {
      "fontSize": "2.5rem",
    },
    "h3": Object {
      "fontSize": "2rem",
    },
    "h4": Object {
      "fontSize": "1.5rem",
    },
    "h5": Object {
      "fontSize": "1.25rem",
    },
    "h6": Object {
      "fontSize": "1rem",
    },
  },
}
`;
